src/util/http/websocket/QueryId.h

```diff
@@ -105,10 +105,10 @@ class QueryRegistry {
   OwningQueryId uniqueId() {
     static thread_local std::mt19937 generator(std::random_device{}());
     std::uniform_int_distribution<uint64_t> distrib{};
-    std::optional<OwningQueryId> result = std::nullopt;
-    while (!result.has_value()) {
+    std::optional<OwningQueryId> result;
+    do {
       result = uniqueIdFromString(std::to_string(distrib(generator)));
-    }
+    } while (!result.has_value());
```

Comment:
What about 
```
while (true) {
 auto result = uniqueIdFromString(...);
 if (result.has_value()) {
   return std::move(result.value());
 }
}
```
(Same number of lines, but more direct to reason about).
I don't have a clear opinion, but as there will be another round I wanted to comment on this.